tests/pubsub/testgossipsub.nim

```diff
@@ -92,4 +91,146 @@ suite "GossipSub":
     expect(CatchableError):
       await gossipSub.rpcHandler(peer, @[byte 1, 2, 3])
 
-    await gossipSub.switch.stop()
+  asyncTest "Peer is disconnected and rate limit is hit when overhead rate limit is exceeded":
+    # Given a GossipSub instance with one peer
+    const topic = "foobar"
+    let
+      (gossipSub, conns, peers) = setupGossipSubWithPeers(1, topic)
+      peer = peers[0]
+      rateLimitHits = currentRateLimitHits("unknown")
+    defer:
+      await teardownGossipSub(gossipSub, conns)
```

Comment:
I think we should move this to a `teardown` block and move all tests that need this to a separate `suite` with that `teardown` block